### ISSUE TITLE
seekZero IRR bug

### ISSUE THREAD

**Issue by giantslogik** [essamjoubori/finance.js#13]
the seekZero function gets into an infinite loop for some  inputs to the IRR function.
Have patched it to abort the calculation after a second

//seekZero seeks the zero point of the function fn(x), accurate to within x \pm 0.01. fn(x) must be decreasing with x.
function seekZero(fn) {
var x = 1;
var ind = new Date().getTime();
while (fn(x) > 0) {
  x += 1;
  if(new Date().getTime()-ind>1000)
	  throw("seekZero failed");
}
while (fn(x) < 0) {
  x -= 0.01
  if(new Date().getTime()-ind>1000)
	  throw("seekZero failed");
}
return x + 0.01;
}

**Comment by gimenete** [comment]
I'm having the same problem. I'll use this IRR implementation: https://gist.github.com/ghalimi/4591338

returns `#NUM!` if the input is not valid, instead of generating an infinite loop.

**Comment by OakBehringer** [comment]
If you'd like to reproduce this, you can use the following: 

```javascript
"use strict";

const Finance = require('financejs');

let finance = new Finance(),
  incomeStream = [998898750, -5478316, -5472758, -5470734, -5624814, -5776428, -5925357, -6071377, -6214268, -6353805, -84877754],
  irr = -1;

try {
  irr = finance.IRR(...incomeStream);
}
catch (e) {
  console.log('IRR Computation ERROR', this.get('label'), incomeStream, e);
}

console.log(irr);```

You will never _catch_ or _console.log_ anything.

**Comment by essamjoubori** [comment]
The bug was fixed in 4.0.1 release. Reopen the issue if you are still facing the issue. Closing this for now.
